### ISSUE TITLE
linux-tfw perfrmance degradation

### ISSUE THREAD

**Issue by krizhanovsky** [tempesta-tech/tempesta#980]
### Description

Nginx shows worse performance on our kernel than on vanilla kernel. I suppose the problem is in our skb page allocator which wastes too much memory, but I'm not sure. Have a look on [fastsocket](http://natsys-lab.blogspot.ru/) - they implement per-core pre-allocated skb pool, so `kmem_cache_alloc()` uses per-cpu cache.

### Environment

2 Nginx instances configured as a web server and non-caching proxy in VM (2 vCPUs, 2GB RAM). Debian 9 with standard kernel and our 4.9.35-tfw.

### Nginx web server

```
ser www-data;
worker_processes auto;
pid /run/nginx.pid;
include /etc/nginx/modules-enabled/*.conf;

events {
    worker_connections   65536;
    use epoll;
    multi_accept on;
    accept_mutex off;
}
worker_rlimit_nofile    1000000;

http {
    keepalive_timeout 600;
    keepalive_requests 10000000;
    sendfile         on;
    tcp_nopush       on;
    tcp_nodelay      on;

    open_file_cache max=1000;
    open_file_cache_valid 30s;
    open_file_cache_min_uses 2;
    open_file_cache_errors off;

    # [ debug | info | notice | warn | error | crit | alert | emerg ]
    # Fully disable log errors.
    error_log /dev/null emerg;

    # Disable access log altogether.
    access_log off;

    server {
	listen 9090 backlog=131072 deferred reuseport fastopen=4096;

        location / {
            root /var/www/html;
        }
        location /nginx_status {
            stub_status on;
        }
    }
}
```

```bash
# cat /var/www/html/index.html 
Ok
```

### Nginx proxy
```
user www-data;
worker_processes auto;
pid /run/nginx.pid;
include /etc/nginx/modules-enabled/*.conf;

events {
	worker_connections 65536;
	use epoll;
	multi_accept on;
	accept_mutex off;
}

http {
	sendfile off;
	tcp_nopush on;
	tcp_nodelay on;
	keepalive_timeout 600;
	keepalive_requests 1000000;

	access_log off;
	# Fully disable log errors.
	error_log /dev/null emerg;
	# error_log /var/log/nginx/error.log;

	gzip off;

	server {
		listen 9000 backlog=131072 deferred reuseport fastopen=4096;
		location / {
			proxy_pass http://127.0.0.1:9090/;
		}
	}
}
```

#### Debian kernel
```
# ulimit -n 65536; ./wrk -c 4096 -t 8 -d 30 http://192.168.100.4:9000/
Running 30s test @ http://192.168.100.4:9000/
  8 threads and 4096 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   111.23ms  151.69ms   1.99s    95.62%
    Req/Sec     1.36k   585.17     4.23k    76.82%
  325883 requests in 30.05s, 75.63MB read
  Socket errors: connect 0, read 12, write 0, timeout 1736
  Non-2xx or 3xx responses: 17505
Requests/sec:  10843.61
Transfer/sec:      2.52MB
```
#### Tempesta kernel
```
# ulimit -n 65536; ./wrk -c 4096 -t 8 -d 30 http://192.168.100.4:9000/
Running 30s test @ http://192.168.100.4:9000/
  8 threads and 4096 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    79.78ms  133.44ms   1.92s    96.05%
    Req/Sec     1.04k     0.89k    4.92k    77.23%
  222928 requests in 30.06s, 51.44MB read
  Socket errors: connect 0, read 43, write 0, timeout 957
  Non-2xx or 3xx responses: 9330
Requests/sec:   7415.66
Transfer/sec:      1.71MB
```

**Comment by krizhanovsky** [comment]
The issue is invalid, the tests were with loaded Tempesta FW in parallel. Tested once more with Tempesta kernel and w/o Tempesta FW show the same performance:
```
 ulimit -n 65536; ./wrk -c 4096 -t 8 -d 30 http://192.168.100.4:9000/
Running 30s test @ http://192.168.100.4:9000/
  8 threads and 4096 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   114.57ms  154.69ms   2.00s    95.21%
    Req/Sec     1.32k   392.08     3.40k    74.96%
  316204 requests in 30.06s, 73.63MB read
  Socket errors: connect 0, read 5, write 0, timeout 1474
  Non-2xx or 3xx responses: 19203
Requests/sec:  10519.25
Transfer/sec:      2.45MB
```
